saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -38,6 +44,15 @@ class Meta:
         error_type_class = ProductError
         error_type_field = "product_errors"
 
+    @classmethod
+    def bulk_action(cls, queryset):
+        products = models.Product.objects.none()
+        for category in queryset:
+            products = products | unpublished_products_before_category_delete(category)
+        product_ids = [product.pk for product in products]
+        super().bulk_action(queryset)
```

Comment:
Should be inside a transaction

Reply:
Not sure if this is part of this PR - it's `BaseBulkMutation`'s reponsibility to ensure that there is a transaction.